lib/internal/streams/destroy.js

```diff
@@ -267,7 +267,7 @@ function constructNT(stream) {
     } else if (err) {
       errorOrDestroy(stream, err, true);
     } else {
-      process.nextTick(emitConstructNT, stream);
+      emitConstructNT(stream);
```

Comment:
This fixes the issue, @ronag I see you added this code, is there a purpose for the next tick?

Reply:
I'm still trying to understand why it was fixed... race condition between who?